File: var/www/cgi-bin/testPOST.py
#!/usr/bin/env python3
import os
import sys
import urllib.parse

def handle_post():
    # Retrieve the content length from the environment
    content_length = int(os.environ.get("CONTENT_LENGTH", 0))

    # Read the POST body from stdin
    post_body = sys.stdin.read(content_length)

    # Parse the POST body
    params = urllib.parse.parse_qs(post_body)

    # Prepare the response body
    response_body = """
    <html>
    <head><title>CGI POST Response</title></head>
    <body>
    <h1>Hello, World!</h1>
    <p>This is a response from a CGI script for a POST request.</p>
    <p>Received POST data:</p>
    <ul>
    """
    for key, values in params.items():
        response_body += f"<li>{key}: {', '.join(values)}</li>\n"

    response_body += """
    </ul>
    </body>
    </html>
    """

    # Calculate the content length of the response body
    response_content_length = len(response_body)

    # Set up headers for the response
    print(f"Status: 200 OK\r\nContent-Type: application/x-www-form-urlencoded\r\nContent-Length: {response_content_length}\r\n\r\n")

    # Output the response body
    print(response_body)

if __name__ == "__main__":
    handle_post()

File: var/www/cgi-bin/testPOST.pl
#!/usr/bin/perl

use strict;
use warnings;
use CGI;

sub handle_post {
    my $cgi = CGI->new;

    # Read the POST data
    my %params = $cgi->Vars;

    # Prepare the response body
    my $response_body = "<html><head><title>POST Request</title></head><body>";
    $response_body .= "<h1>Received POST data</h1>";
    $response_body .= "<ul>";

    # Iterate over all POST parameters
    foreach my $key (sort keys %params) {
        my $value = $params{$key};
        $response_body .= "<li><strong>$key</strong>: $value</li>";
    }

    $response_body .= "</ul>";
    $response_body .= "</body></html>";

    # Calculate the content length of the response body
    my $response_content_length = length($response_body);

    # Set up headers for the response
    print "Status: 200 OK\r\n";
    print "Content-Type: text/html\r\n";
    print "Content-Length: $response_content_length\r\n";
    print "\r\n";

    # Output the response body
    print $response_body;
}

sub main {
	handle_post();
}

main();
